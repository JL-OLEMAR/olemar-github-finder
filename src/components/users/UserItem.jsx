import { Link } from 'react-router-dom'
import PropTypes from 'prop-types'

export function UserItem({ user }) {
  return (
    <div className='card shadow-md compact side bg-base-100'>
      <div className='flex-row items-center space-x-4 card-body'>
        <div>
          <div className='avatar'>
            <div className='rounded-full w-14 h-14 shadow'>
              <img alt={`${user.login} Profile`} src={user.avatar_url} />
            </div>
          </div>
        </div>

        <div>
          <h2 className='card-title'>{user.login}</h2>
          <Link
            className='text-opacity-40 text-base-content'
            to={`/user/${user.login}`}
          >
            Visit Profile
          </Link>
        </div>
      </div>
    </div>
  )
}

UserItem.propTypes = {
  user: PropTypes.object.isRequired
}
