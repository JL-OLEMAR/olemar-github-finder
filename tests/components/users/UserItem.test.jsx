import { MemoryRouter } from 'react-router-dom'
import { render, screen } from '@testing-library/react'

import { UserItem } from '../../../src/components/users/UserItem.jsx'

describe('Tests in <UserItem />', () => {
  const user = {
    login: 'JL-OLEMAR',
    avatar_url: 'https://avatars.githubusercontent.com/u/71868068?v=4'
  }

  test('should match the snapshot', () => {
    const { container } = render(
      <MemoryRouter>
        <UserItem user={user} />
      </MemoryRouter>
    )

    expect(container).toMatchSnapshot()
  })

  test('should show the image with URL and ALT correctly', () => {
    render(
      <MemoryRouter>
        <UserItem user={user} />
      </MemoryRouter>
    )

    const { src, alt } = screen.getByRole('img')

    expect(src).toBe(user.avatar_url)
    expect(alt).toBe(`${user.login} Profile`)
  })

  test('should show the title with the user login correctly', () => {
    render(
      <MemoryRouter>
        <UserItem user={user} />
      </MemoryRouter>
    )

    expect(screen.getByText(user.login)).toBeTruthy()
  })

  test('should show the href of the link correctly', () => {
    render(
      <MemoryRouter>
        <UserItem user={user} />
      </MemoryRouter>
    )

    const linkElement = screen.getByRole('link').getAttribute('href')

    expect(linkElement).toBe(`/user/${user.login}`)
  })
})
