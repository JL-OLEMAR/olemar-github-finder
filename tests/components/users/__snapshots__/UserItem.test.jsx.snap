// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <UserItem /> should match the snapshot 1`] = `
<div>
  <div
    class="card shadow-md compact side bg-base-100"
  >
    <div
      class="flex-row items-center space-x-4 card-body"
    >
      <div>
        <div
          class="avatar"
        >
          <div
            class="rounded-full w-14 h-14 shadow"
          >
            <img
              alt="JL-OLEMAR Profile"
              src="https://avatars.githubusercontent.com/u/71868068?v=4"
            />
          </div>
        </div>
      </div>
      <div>
        <h2
          class="card-title"
        >
          JL-OLEMAR
        </h2>
        <a
          class="text-opacity-40 text-base-content"
          href="/user/JL-OLEMAR"
        >
          Visit Profile
        </a>
      </div>
    </div>
  </div>
</div>
`;
